Java Record support (#365)
jakartaee/jsonb-api: spec/src/main/asciidoc/jsonb.adoc

JSON object values are deserialized into an implementation of `java.util.Map<String, Object>` with a predictable iteration order.

=== Java Class and Record

Any class instance passed to a deserialization operation must have a public or protected no-argument constructor. Implementations SHOULD throw an error if this condition is not met. This limitation does not apply to serialization operations, as well as to classes which specify explicit instantiation methods as described in section 4.5.

> Any record instance passed to a deserialization operation must have only default generated constructor. Implementations SHOULD throw an error if this condition is not met. This limitation does not apply to serialization operations, as well as to records which specify explicit instantiation methods as described in section 4.5.

Reviewer: To allow compact constructor (see prior comment),

```suggestion
Any record instance passed to a deserialization operation must have its default generated constructor (and optionally a compact constructor that does not reassign any record component values) as its only constructor. Implementations SHOULD throw an error if this condition is not met. This limitation does not apply to serialization operations, as well as to records which specify explicit instantiation methods as described in section 4.5.
```

If `m = 1`, then the most specific parameter type MUST be derived from the given bound.

> If is class, record or interface, the most specific parameter type MUST be the class, record or interface.

Reviewer: Minor grammar correction:

```suggestion
If it is a class, record or interface, the most specific parameter type MUST be the class, record or interface.
```

=== Customizing Property Names

There are two standard ways how to customize serialization of field or record component (or JavaBean property) to JSON document. The same applies to deserialization. The first way is to annotate field or record component (or JavaBean property) with `jakarta.json.bind.annotation.JsonbProperty` annotation. The second option is to set `jakarta.json.bind.config.PropertyNamingStrategy`.

==== jakarta.json.bind.annotation.JsonbTransient

JSON Binding implementations MUST NOT process fields, record components, JavaBean properties or types annotated with `jakarta.json.bind.annotation.JsonbTransient`.

`JsonbTransient` annotation is mutually exclusive with all other JSON Binding defined annotations. Implementations must throw `JsonbException` in the following cases:

* Class field or record component is annotated with `@JsonbTransient`
** Exception must be thrown when this field, record component, getter or setter is annotated with other JSON Binding annotations.
* Getter is annotated with `@JsonbTransient`

> ** Exception is thrown if when the field, record component or this getter are annotated with other JSON Binding annotations. Exception is not thrown if JSON Binding annotations are presented on the setter.

Reviewer: Minor grammar corrections:

```suggestion
** An exception is thrown if the field, record component or this getter are annotated with other JSON Binding annotations. An exception is not thrown if JSON Binding annotations are present on the setter.
```
